src/ImageSharp.Generators/ImageExtensionsSaveGenerator.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace ImageSharp.Generators;
+
+[Generator(LanguageNames.CSharp)]
+internal class ImageExtensionsSaveGenerator : IIncrementalGenerator
+{
+    private const string FileHeader = @"// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+// <auto-generated />";
+
+    private static readonly string[] ImageFormats =
```

Comment:
Wouldn't it be better to have some sort of attribute on the encoder type that can be used to generate the methods instead?

That would make it more generator worthy would it not @Sergio0694 ?

It would be amazing if we could add that to a base class (ImageEncoder See #2269) and have these methods generated automatically without us ever having to touch this file again.

Reply:
Yup, this would be possible and probably a better suited scenario for a Source Generator. You'd still have the issues around local projects with SGs which @Sergio0694 mentioned in https://github.com/SixLabors/ImageSharp/pull/2237#issuecomment-1252359522 and would need to decide if this is okay for you.

Reply:
> Not to mention, using source generators with a local project reference is unsupported, and has broken tooling

This one? I'm still blown away by that. How does someone test such a thing if locally referencing the project is unsupported? It seems incredibly short sighted.

Reply:
Yeah, to be honest I haven't noticed this as much. I rarely clean my repository like that.
I have multiple SGs in a local project at work and only get red squiggly lines when modifying the SG code, which rarely happens. Even then closing and reopening the solution fixes it. Other than that, all is good.